Design/Instrument-2.0/Detector_Scan_NeXus_Saving.md

```diff
@@ -0,0 +1,139 @@
+
+The NeXus proposal for scanning allows the start and end positions and rotations to be different for a scan interval. Logically you might then interpolate between the start and end values for something that is continuously moving. This is not supported in Mantid, during a scan interval the detector/pixel is stationary.
+
+### Component Scanning
+
+_Based on discussion with Simon Heybrock._
+
+To resolve the differences between the NeXus proposal and Mantid one option is to allow components to scan. This is similar to how `RectangularDetector` works. The changes required would be:
+
+* Add the scan interface in `DetectorInfo` to `ComponentInfo`
+  * E.g. in addition to `ComponentInfo::setPosition(detIndex)` we have `ComponentInfo::setPosition({detIndex, timeIndex})`
+  * A `ComponentInfo::setPosition()` call updates all of the child positions (same for rotations)
+* For setting positions, rotations or mask that can change with the time index, both `ComponentInfo` and `DetectorInfo` will require a check to see if the parent is scanning
+  * If the parent is scanning it should throw
+* The in memory representation is the same as before - the absolute position and rotation of all pixels are stored
+
+The performance impact of this is that every call to `DetectorInfo::setPosition()` and similar will have an extra branch, even for non-scanning workspaces.
+
+An alternative approach is to have something similar to `RectangularDetectors`. Here instead of storing the scan positions for each detector/pixel we would just store the position for the parent. Positions would need to computed on the fly to get the actual position for each detector/pixel. This would have the advantage of reducing memory use*, at the cost of more CPU work. This would be a step away from some of the Instrument 2.0 design principles.
+
+*Note that the memory overhead due to scanning is not the main issue for the ILL instruments, addressing the overhead from using histograms with a  single count (no ToF) is a higher priority.
+
+#### Outstanding Question
+
+* Should pixels be allowed to scan, given that the parenting is not scanning?
+  * Possible use cases for this would be tubes that might change in time, for example temperature, during a scan. The calibration might then be time dependent.
+  * If the answer to this is that we do need per-pixel scans then how can we support this in the NeXus file? A different proposal may be required.
+  * Do we then need a mechanism in Mantid to define which components can scan, and which can not?
+
+This is the key question to answer - if we restrict the scans in this way we should be able to use the NeXus Instrument proposal as it stands. If we do not impose this restriction we will likely need something to allow the values in `x_pixel_offset` and similar to be set per time interval.
```

Comment:
`NXlog` can be used anywhere that time dependence is required for a dataset. So it would certainly be possible to use `NXlog`s for the pixel offsets if required.

Reply:
Would this require an `NXlog` for each pixel offset, or can you give an array of values?

Reply:
An array is fine, from the documentation: "Array of logged value, such as temperature. If this is a single value the dimensionality is nEntries. However, NXlog can also be used to store multi dimensional time stamped data such as images. In this example the dimensionality of values would be value[nEntries,xdim,ydim]."

Reply:
This is good news then - this fills in the gap I thought was missing. I'll make some updates to the doc to mention this.